specification/src/chapter2.adoc

```diff
@@ -2,9 +2,600 @@
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| All isolated software that has to be trusted by other software on the same system MUST be measured and verified.
+|===
+
+Verification ensures only software from trusted parties is installed.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| All isolated software that provides services to remote reliant parties MUST be at least measured.
+|===
+
+Software that is at least measured can be attested by remote reliant parties. To ensure supply chain integrity it is recommended that it is also verified locally before installation.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| Software authorization MUST be rooted in immutable boot code.
+|===
+
+Immutable boot code forms part of the immutable hardware security platform identified by the hardware security platform identity and is the start of a local trust chain on a system. It is typically ROM code, or locked on-chip flash.
+
+If a system implements a hardware root of trust, then the HW root of trust hosts the immutable boot code. Otherwise the immutable boot code is typically the first code that executes on an application processor immediately after reset. 
+
+==== Secure updates
+
+Over time, any non-immutable component may need updates to address vulnerabilities or functionality improvements. An update can concern software/firmware, microcode, or any other updatable element on a system.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| All software on a system which is not immutable MUST be updatable.
+|===
+
+Immutable software includes at least immutable boot code. Some trusted subsystems can also include immutable software to meet specific security certification requirements defined by a governing body. 
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| An update of a software component SHOULD restart all other dependant software components.
+|===
+
+A software update changes the attested security state of the affected component, and can affect whether the system is still considered trustworthy or not by a reliant party. A restart forces determination of a fresh attested state following the update. 
+
+Alternatively, support for live updates may be part of the already attested trust contract between the reliant party and the system, given suitable governance. In which case a restart may not be required.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| An update of trusted security services MUST only take effect following a system reset.
```

Comment:
most systems now want to update without a full reset.

Reply:
In server land maybe :-) 

Ok, I will review this section and suggest an update. We are basically talking about either local boot rooted in immutable boot code, where at least the trusted security services must be measure and verified; or host attestation where system boot only is measured, but the the host must be attested before launching hosted services.

Reply:
I have updated the "Authorized software" section above to prepare the ground. Please review.

Will tweak the update section tomorrow.

Reply:
Also in other lands like industrial control systems that run mission critical processes.

Reply:
Pushed an update to the system update section as well. Please review.